src/casp/scripts/bq_ops/sql_templates/prepare_curriculum/prepare_feature_summary.sql.mako

```diff
@@ -9,7 +9,7 @@ select f.original_feature_id,
 from `${project}.${dataset}.cas_raw_count_matrix` m
 join `${project}.${dataset}.cas_feature_info` f on (m.cas_feature_index = f.cas_feature_index)
 join `${project}.${dataset}.cas_cell_info` c on (m.cas_cell_index = c.cas_cell_index)
-% if "dataset_id__eq" in filter_statements.keys() or "dataset_id__in" in filter_statements.keys():
+% if "dataset_id" in " ".join(filter_statements.keys()):
```

Comment:
What is accomplished by the join here?

Reply:
I need to check if dataset_id appears within the filter statements. Filter statements is a dictionary where the keys represent different filters. Filters represent a string with `{column_name}__{filter_name}` (you can find more info on how filters are generated in [function that renders where clause](https://github.com/cellarium-ai/cellarium-cloud/pull/195/files#diff-4f00b415c6a4deb334cf90b14f78db8424e9a0071fddbfb8e2a550d97f3d4271L111). I wanted a simple method to check if dataset_id could be a substring within any filter key. I'm doing this by using join instead of checking if it’s in the list, as the key for filtering dataset_id appears to be different every time depending on `filter_name`. Previously, it was just checking `if dataset_id__eq in filter_statements.keys() or dataset_id__in in filter_statements.keys()`. However, I realized there are other options like `dataset_id__not_in` and `dataset_id__not_eq` and this if statement doesn't work with them.

So, rather than checking all possible filters, I thought checking if dataset_id is a substring within filter_statements would be simpler, this way it'd catch it no matter what filter name is.

Reply:
Ooooh okay, that makes sense to me